RFCs/0070-Fn::Size.md

```diff
@@ -0,0 +1,104 @@
+# RFC for Fn::Size
+
+* **Original Author(s):**: @juegong2
+* **Tracking Issue**: https://github.com/aws-cloudformation/cfn-language-discussion/issues/70
+
+# Summary
+
+We will support an intrinsic function called `Fn::Size` that will return the number of elements in a given list.
+
+# Motivation
+
+CloudFormation users may want to get the number of elements in a given list to set certain resource parameters or generate conditions. It may also help to check if index is out of bound for !Select function. This RFC proposes to support the use case of getting the number of elements of a given list.
+
+* Reference:
+  * https://github.com/aws-cloudformation/cfn-language-discussion/issues/61
+
+# Example(s)
+
+Use `Ref` on a Parameter of type `CommaDelimitedList` or `List<Number>`
+
+```json
+{
+  "Parameters": {
+    "InstanceTypes": {
+      "Type": "CommaDelimitedList"
+    },
+    "NumberList": {
+      "Type": "List<Number>"
+    }
+  }
+  "Conditions": {
+    "Has3InstanceTypes": {
+      "Fn::Equals": [
+        {
+          "Fn::Size": {"Ref": "InstanceTypes"}
+        },
+        3
+      ]
+    },
+    "Has2NumbersInList": {
+      "Fn::Equals": [
+        {
+          "Fn::Size": {"Ref": "NumberList"}
+        },
+        2
+      ]
+    }
+  }
+}
+```
+
+Use a hardcoded list
+```json
+{
+  "Conditions": {
+    "Has3InstanceTypes": {
+      "Fn::Equals": [
+        {
+          "Fn::Size": [ "m5.large", "m5.xlarge" ]
+        },
+        3
+      ]
+    }
+  }
+}
+```
+
+# Limitation
+
+`Fn::Size` will not work with lists that are dynamically generated at provisinoing time. For example, assuming MyBucket is the logical id of a AWS::S3::Bucket resource, the following is not supported:
+```json
+{
+  "Fn::Size": { "Fn::Split": [":", { "Fn::GetAtt": [ "MyBucket", "Arn" ] } ] }
+}
+```
+because the ARN of MyBucket is not known until during provisioning.
+
+# Details
+
+`Fn::Size` is an intrinsic function that takes in a list and returns the size of the given list. The list could be:
+
+* hardcoded list in template
+* array generated by intrinsic functions like `Fn::Split` of hardcoded parameters or of `Ref` to parameters
```

Comment:
```suggestion
* array generated by intrinsic functions like `Fn::Split` of hardcoded values or of `Ref` to parameters
```

Reply:
here I meant parameter of an intrinsic function, not parameter of the template. But yeah it will cause confusion :(